src/main/java/de/zalando/aruha/nakadi/repository/db/SubscriptionDbRepository.java

```diff
@@ -0,0 +1,82 @@
+package de.zalando.aruha.nakadi.repository.db;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.zalando.aruha.nakadi.domain.Subscription;
+import de.zalando.aruha.nakadi.exceptions.DuplicatedSubscriptionException;
+import de.zalando.aruha.nakadi.exceptions.InternalNakadiException;
+import de.zalando.aruha.nakadi.exceptions.NoSuchSubscriptionException;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Set;
+
+import static com.google.common.collect.Sets.newTreeSet;
+
+public class SubscriptionDbRepository extends AbstractDbRepository {
+
+    private final SubscriptionMapper rowMapper = new SubscriptionMapper();
+
+    public SubscriptionDbRepository(final JdbcTemplate jdbcTemplate, final ObjectMapper objectMapper) {
+        super(jdbcTemplate, objectMapper);
+    }
+
+    public void saveSubscription(final Subscription subscription) throws InternalNakadiException,
```

Comment:
And again, I don't like the idea of storing in subscription itself, because of:
 - at least 2 additional checks
 - id is not a part of subscription that was initialy created by user (and that's the difference from EventType)
 - not logical to store it twice